#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: sync builtin CDPATH documentation with man page

--- builtins/reserved.def~	2002-04-04 21:52:41.000000000 +0200
+++ builtins/reserved.def	2004-10-16 18:59:06.000000000 +0200
@@ -145,9 +145,9 @@
 $DOCNAME variable_help
 $SHORT_DOC variables - Some variable names and meanings
 BASH_VERSION    Version information for this Bash.
-CDPATH          A colon separated list of directories to search
-		when the argument to `cd' is not found in the current
-		directory.
+CDPATH          The search path for the `cd' command. A colon-separated list
+		of directories in which the shell looks for destination
+		directories specified by the `cd' command.
 GLOBIGNORE	A colon-separated list of patterns describing filenames to
 		be ignored by pathname expansion.
 #if defined (HISTORY)
